internal/urlstorage/urlstorage.go

```diff
@@ -0,0 +1,100 @@
+package urlstorage
+
+import (
+	"context"
+	"errors"
+	"shorter/internal/domain"
+	generateRandomId "shorter/pkg"
+	"sync"
+)
+
+type repository interface {
+	Save(values domain.URLS, userID int) error
+	GetURL(shortURL string) (string, error)
+	GetInitialData() (domain.URLStorage, error)
+	SaveBatch(ctx context.Context, values []domain.URLS, userID int) error
+	GetUserURLs(userID int, serverAdr string) ([]domain.UserURLs, error)
```

Comment:
```suggestion
	Save(ctx context.Context, values domain.URLS, userID int) error
	GetURL(ctx context.Context, shortURL string) (string, error)
	SaveBatch(ctx context.Context, values []domain.URLS, userID int) error
	GetUserURLs(ctx context.Context, userID int, serverAdr string) ([]domain.UserURLs, error)
```
Интерфейс должен содержать только общие методы для реализации функионала. Не очень понятно, что должен будет вернуть файловый и БД интерфейс в GetInitialData

В операции с БД необходимо пробрасывать контекст от запроса. Во всех остальных реализациях можно зашунтировать получение этого аргумента (`_ context.Context`)